src/receiver-ui.py

```diff
@@ -0,0 +1,119 @@
+import ipaddress, subprocess
+from tkinter import *
+from tkinter import filedialog, messagebox
+from receiver import Receiver
+
+
+def register():
+    receiver.display_name = display_name_entry.get()
+    receiver.serial_number = serial_number_entry.get()
+    return_message = receiver.register()
+    if return_message == "Decoder already registered!":
+        messagebox.showerror("Error", return_message)
+    else:
+        messagebox.showinfo("Info", return_message)
+
+
+def receive():
+    ip = ip_entry.get()
+    port = port_entry.get()
+    if not is_valid_ip(ip):
+        messagebox.showerror("Error", "Invalid IP address.")
+        return
+    if not is_valid_port(port):
+        messagebox.showerror("Error", "Invalid port.")
+        return
+    subprocess.Popen(
+        [
+            "ffplay",
+            "-autoexit",
+            "-v",
+            "warning",
+            "-stats",
+            f"srt://{ip}:{port}?mode=listener",
+        ]
+    )
+
+
+def is_valid_ip(ip):
+    try:
+        ipaddress.ip_address(ip)
+    except ValueError:
+        return False
+    return True
+
+
+def is_valid_port(port):
+    try:
+        port = int(port)
+        if port >= 1 and port <= 65535:
+            return True
+        else:
+            return False
+    except ValueError:
+        return False
+
+
+root = Tk()
+root.title("Switchboard - Sample Receiver")
+root.geometry("800x400")
+root.iconphoto(True, PhotoImage(file=r"public\bean.png"))
+receiver = Receiver()
+default_font = ("TkDefaultFont", 12)
+
+# Registration section elements
+registration_label_frame = LabelFrame(
+    root, text="Registration", font=default_font, borderwidth=4
+)
+display_name_label = Label(
+    registration_label_frame, text="Display Name", font=default_font, width=20
+)
+display_name_entry = Entry(registration_label_frame, width=30, font=default_font)
+display_name_entry.insert(0, receiver.display_name)
+serial_number_label = Label(
+    registration_label_frame, text="Serial Number", font=default_font, width=20
+)
+serial_number_entry = Entry(registration_label_frame, width=30, font=default_font)
+serial_number_entry.insert(0, receiver.serial_number)
+register_button = Button(
+    registration_label_frame,
+    text="Register",
+    font=default_font,
+    bg="#57A834",
+    width=20,
+    command=register,
+)
+
+# Listening Instructions section elements
+listening_label_frame = LabelFrame(
+    root, text="Listening Instructions", font=default_font, borderwidth=4
+)
+ip_label = Label(listening_label_frame, text="IP", font=default_font, width=10)
+ip_entry = Entry(listening_label_frame, width=40, font=default_font)
```

Comment:
I asked Maxim what the IP field is used for in the receiver, he said you need to enter the IP address from which you are receiving (the ip address of the sender).